### PR TITLE
tokio: remove wildcard in match pattern

### PR DESCRIPTION
**Pull request by wathenjiang**
When we use wildcard `*` in match pattern, it can lead to potential errors.

```rust
enum MyEnum{
    TypeA,
    TypeB,
    TypeC,
}

fn logic(e :MyEnum){
    use MyEnum::*;

    match e {
        TypeA =>  println!("A"),
        TypeB =>  println!("B"),
        TypeC =>  println!("C"),
    }
}
```

If we delete `TypeC`，add add `TypeD`

```
enum MyEnum{
    TypeA,
    TypeB,
    TypeD,
}

fn logic(e :MyEnum){
    use MyEnum::*;

    match e {
        TypeA =>  println!("A"),
        TypeB =>  println!("B"),
        TypeC =>  println!("C"),
    }
}
```

The Rust compiler takes `TypeC` as a named variables,  not a variant of the enum, and the unexpected error occurs.

So we should never use such as `use MyEnum::*;` wildcard in Rust code.

### CODE REVIEW

**Comment by Darksonn** [comment]
That would trigger a warning because TypeC is not snake case, which would fail our build.

**Comment by wathenjiang** [comment]
> That would trigger a warning because TypeC is not snake case, which would fail our build.

Yes, the CI is useful for it. But if we do this, we can avoid such error before CI, and it may have better code smell.

File: tokio-macros/src/entry.rs [hunk 1]
```diff
@@ -127,21 +127,20 @@ impl Configuration {
 
     fn build(&self) -> Result<FinalConfig, syn::Error> {
         let flavor = self.flavor.unwrap_or(self.default_flavor);
-        use RuntimeFlavor::*;
 
         let worker_threads = match (flavor, self.worker_threads) {
-            (CurrentThread, Some((_, worker_threads_span))) => {
+            (RuntimeFlavor::CurrentThread, Some((_, worker_threads_span))) => {
```
Comment (Darksonn) [inline tokio-macros/src/entry.rs:132]: I don't mind dropping the import for this case, but I don't want to drop it for `Poll`, `Result`, `Ok`, or `Ordering`. That's too much noise.

Comment (wathenjiang) [reply tokio-macros/src/entry.rs:132]: > I don't mind dropping the import for this case, but I don't want to drop it for `Poll`, `Result`, `Ok`, or `Ordering`. That's too much noise.

Some kinds of enum of them, such as `Result`, `Ordering` might not be changed in the future, unless the std of rust make some broken changes. I reserve my opinion on these enums. But I believe some enums which is defiend in internal code of tokio might change in the future.

We can use `use MyEnum as E` to reduce the noise.

Comment (wathenjiang) [reply tokio-macros/src/entry.rs]: I have made some changes, allowing the wildcard in Result and Ordering for reduce the noise. And keep the other changes.

File: tokio/src/runtime/handle.rs [hunk 2]
```diff
@@ -604,20 +604,20 @@ enum TryCurrentErrorKind {
 
 impl fmt::Debug for TryCurrentErrorKind {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        use TryCurrentErrorKind::*;
+        use TryCurrentErrorKind as E;
         match self {
-            NoContext => f.write_str("NoContext"),
-            ThreadLocalDestroyed => f.write_str("ThreadLocalDestroyed"),
+            E::NoContext => f.write_str("NoContext"),
+            E::ThreadLocalDestroyed => f.write_str("ThreadLocalDestroyed"),
```
Comment (Darksonn) [inline tokio/src/runtime/handle.rs:610]: Why E? Should this not be a K for kind?

I'm also okay with `TryCurrentErrorKind::` here.

Comment (wathenjiang) [reply tokio/src/runtime/handle.rs:610]: E means Eroor, and I may have been influenced by to golang :). I believe we in Rust the error have a kind, which can refer to: https://doc.rust-lang.org/std/io/enum.ErrorKind.html, and the use of `TryCurrentErrorKind ::` here is not tedious, so let us just use `TryCurrentErrorKind ::`.

**Comment by wathenjiang** [comment]
The CI has passed, it looked like the last CI gone wrong.

**Review by Darksonn — APPROVED** [review]
Thanks.

**Pull request merged**